src/App.tsx

```diff
@@ -1,26 +1,51 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { useEffect, useState } from 'react';
+import { getTodos } from './api';
+import { todosSlice } from './features/todos';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  const [error, setError] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setError('');
+    setIsLoading(true);
+
+    getTodos()
+      .then(data => {
+        dispatch(todosSlice.actions.setTodos(data));
+      })
+      .catch(() => setError('Oops, something went wrong'))
+      .finally(() => setIsLoading(false));
+  }, [dispatch]);
+
+  return (
+    <>
+      <div className="section">
+        <div className="container">
+          <div className="box">
+            <h1 className="title">Todos:</h1>
+
+            <div className="block">
+              <TodoFilter />
+            </div>
 
-          <div className="block">
-            <Loader />
-            <TodoList />
+            <div className="block">
+              {isLoading && <Loader />}
+              {!isLoading && !error && <TodoList />}
+              {!isLoading && error}
```

Comment:
Consider displaying the error message in a more user-friendly way, such as using a styled component or an alert box, instead of rendering it directly.